vaadin-time-picker-flow-parent/vaadin-time-picker-flow-integration-tests/src/test/java/com/vaadin/flow/component/timepicker/tests/TimePickerDetachAttachPageIT.java

```diff
@@ -15,36 +15,61 @@
  */
 package com.vaadin.flow.component.timepicker.tests;
 
+import com.vaadin.flow.component.timepicker.testbench.TimePickerElement;
+import com.vaadin.testbench.TestBenchElement;
 import com.vaadin.tests.AbstractComponentIT;
 import com.vaadin.flow.testutil.TestPath;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
-import org.openqa.selenium.By;
 import org.openqa.selenium.Keys;
-import org.openqa.selenium.WebElement;
 
 /**
  * Integration tests for attaching / detaching time picker.
  */
 @TestPath("vaadin-time-picker/time-picker-detach-attach")
 public class TimePickerDetachAttachPageIT extends AbstractComponentIT {
 
-    @Test
-    public void clientSideValidationIsOverriddenOnAttach() {
+    TestBenchElement toggleAttach;
+    TestBenchElement setValue;
+    TestBenchElement setLocale;
+
+    @Before
+    public void init() {
         open();
+        toggleAttach = $("button").id("toggle-attached");
+        setValue = $("button").id("set-value");
+        setLocale = $("button").id("set-california-locale");
+    }
 
+    @Test
+    public void clientSideValidationIsOverriddenOnAttach() {
         assertTimePickerIsValidOnTab();
 
         // Detaching and attaching time picker
-        WebElement toggleAttach = findElement(By.id("toggle-attached"));
         toggleAttach.click();
         toggleAttach.click();
 
         assertTimePickerIsValidOnTab();
     }
 
+    @Test
+    public void formatShouldRespectLocaleAfterDetachAndReattach() {
```

Comment:
This PR also fixes a bug in the TimePicker where the time format from the locale would not be reapplied when detaching and reattaching. See changes in `TimePicker.onAttach`.